cmake/onnxruntime_providers.cmake

```diff
@@ -690,6 +693,98 @@ if (onnxruntime_USE_NUPHAR)
   install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/providers/nuphar  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime/core/providers)
 endif()
 
+if (onnxruntime_USE_OPENCL)
+  add_definitions(-DUSE_OPENCL=1)
+
+  # Implement simple compiler driver to help
+  #
+  # a.h--------┐
+  # b.h--------┤
+  # kernel.cl -┴-> kernel.cl.inc -┐
+  #                op.cc ---------┴-> op.o
+  #
+  # embed.py is responsible for transforming *.cl and related header files to
+  # *.cl.inc, This enables us easily embedding the kernel src into an object
```

Comment:
nit: embed.py -> onnxruntime/core/providers/opencl/embed.py so it's clear where the script is